tests/common/plugins/sanity_check/checks.py

```diff
@@ -557,3 +561,122 @@ def _check():
 
         return check_results
     return _check
+
+@pytest.fixture(scope="module")
+def check_secureboot(duthosts, request):
+    """
+    Check if the file change in rw folder is as expected when secureboot feature enabled
+    If the file change is only for test, not for product, please add the change in the default_allowlist below
+    """
+
+    default_allowlist = [ r".*\.pyc" ]
+    cache_location = 'secureboot_sanity_check'
+    module = request.module.__name__
+
+    def _read_config_by_dut(duthost):
+        results = {}
+
+        # Check if secure boot enabled
+        check_secureboot_cmd = r"grep -q 'secure_boot_enable=y' /proc/cmdline && echo y"
+        shell_result = duthost.shell(check_secureboot_cmd)
+        if shell_result['stdout'].strip() != 'y':
+            logger.info("Skipped to check secure boot for dut %s, since the secure boot is not enabled" % duthost.hostname)
+            return results
+
+        # Read the allowlist
+        allowlist = []
+        results['allowlist'] = allowlist
+        read_allowlist_cmd = r"IMAGE=$(sed 's#.* loop=\(.*\)/.*#\1#' /proc/cmdline); unzip -p /host/$IMAGE/sonic.swi allowlist_paths.conf"
+        shell_result = duthost.shell(read_allowlist_cmd)
+        stdout = shell_result['stdout']
+        for line in stdout.split('\n'):
+            line = line.strip()
+            if len(line) > 0:
+                allowlist.append(line)
+        logger.info("Read %d allowlist settings from dut %s" % (len(allowlist), duthost.hostname))
+
+        # Read the rw files
+        rw_files = {}
+        results['rw'] = rw_files
+        ls_rw_files_cmd = r"IMAGE=$(sed 's#.* loop=\(.*\)/.*#\1#' /proc/cmdline); find /host/$IMAGE/rw -type f -exec md5sum {} \; | sed -E 's#/host/[^/]+/rw##g'"
+        shell_result = duthost.shell(ls_rw_files_cmd)
+        stdout = shell_result['stdout']
+        for line in stdout.split('\n'):
+            line = line.strip()
+            if len(line) > 33:
+                filename = line[33:].strip()
+                rw_files[filename] = line[:32] #md5sum
+        logger.info("Read %d rw files from dut %s" % (len(rw_files), duthost.hostname))
+
+        return results
+
+    def _read_configs():
+        results = {}
+        for duthost in duthosts:
+            config = _read_config_by_dut(duthost)
+            if config:
+              results[duthost.hostname] = config
```

Comment:
It looks like the indention is still incorrect.

Reply:
Fixed, thanks